### DISCUSSION TITLE
Provide a sensor showing which mode the Car has selected when "Climate" is activated

### DISCUSSION THREAD

**Discussion by davebarrett1959** [bimmerconnected/bimmer_connected#595] [Feature request] [answered]
The following is based on findings with a G05 hybrid (X5 45e) with idrive 7.
When the "Climate" switch is enabled and sent to the car the car decides which of 3 modes it will activate. These are -

1. Fan only
2. Air conditioning
3. Heating

It appears there is no way to control this either remotely or via the in car settings. It some cases (maybe only on idrive 7?) it incorrectly sets this and the fix is to disable and re-enable the climate. This applies either via HA or the MyBMW app.

In the MyBMW app the option set by the car is shown by means of wavy coloured lines running through the car image.  (White for fan only, blue for Air Conditioning and red for Heating). As the car decides this it must send back some form of result code to the server and hence the app.

Could this result code be added to the HA integration as a new sensor so that when climate is triggered in HA it will be updated and one can see if it has set the intended mode.

Many Thanks

**Comment by brave0d** [comment]
Hi, this would be similar or even the same as: https://github.com/bimmerconnected/bimmer_connected/discussions/576

**Reply by rikroe** [reply, marked as answer]
Implemented as [home-assistant/core#110287](https://github.com/home-assistant/core/pull/110287) got merged.
